common/scala/src/main/scala/whisk/core/containerpool/AkkaContainerClient.scala

```diff
@@ -162,7 +162,7 @@ protected class AkkaContainerClient(
         } else {
           //ignore the tail (MUST CONSUME ENTIRE ENTITY!)
           //captured string MAY be larger than the max response, so take only maxResponse bytes to get the exact length
-          tail.runWith(Sink.ignore).map(_ => truncatedResponse.take(maxResponse.toBytes.toInt).utf8String)
+          tail.runWith(Sink.ignore).map(_ => truncatedResponse.take(maxResponse.toBytes.toInt - 1024).utf8String)
```

Comment:
For truncation, we need to account for bytes that are added to the message when the Kafka producer serializes the message.